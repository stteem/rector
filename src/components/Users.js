import React, { Component } from "react";

export default class Users extends Component {
  constructor(props) {
    super(props);
    this.state = { data: [] };
  }

  componentDidMount() {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then(response => {
        // make sure to check for errors
        return response.json();
      })
      .then(json => {
        this.setState(() => {
          return { data: json };
        });
      })
      .catch((error) => { return error; });
  }
  render() {
    return (
      <>
      <center>
          <h2>Guests And Users</h2>
      </center>
      
      <ul>
        {this.state.data.map(user => (
          <li key={user.name}>{user.name}</li>
        ))}
      </ul>
      </>
    );
  }
}