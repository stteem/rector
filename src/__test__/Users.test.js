import React from "react";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import { create } from 'react-test-renderer';

import Users from "../components/Users";

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("User component", () => {
  it("it shows a list of users", async () => {
    const fakeResponse = [{ name: "John Doe" }, { name: "Kevin Mitnick" }];

    jest.spyOn(window, "fetch").mockImplementation(() => {
      const fetchResponse = {
        json: () => Promise.resolve(fakeResponse)
      };
      return Promise.resolve(fetchResponse);
    });

    await act(async () => {
      render(<Users />, container);
    });

    const header = container.querySelector('h2');
    const list = container.querySelector('ul');

    expect(header.textContent).toBe("Guests And Users");
    expect(list.textContent).toBe("John DoeKevin Mitnick");

    window.fetch.mockRestore();
  });




  it('takes snapshot of state changes', () => {

    const fakeResponse = [{ name: "John Doe" }, { name: "Kevin Mitnick" }];

    jest.spyOn(window, "fetch").mockImplementation(() => {
      const fetchResponse = {
        json: () => Promise.resolve(fakeResponse)
      };
      return Promise.resolve(fetchResponse);
    });

    let component;
    act(() => {
      component = create(<Users />);
    }); 

    expect(component.toJSON()).toMatchSnapshot();

    window.fetch.mockRestore();
  });
});