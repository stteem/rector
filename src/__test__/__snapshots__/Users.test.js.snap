// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component takes snapshot of state changes 1`] = `
Array [
  <center>
    <h2>
      Guests And Users
    </h2>
  </center>,
  <ul />,
]
`;
